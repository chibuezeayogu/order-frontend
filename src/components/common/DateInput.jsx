import React from 'react'

const DateInput = ({ value, onChange, text }) => {
  return (
    <input
      type="date"
      className="form-control rounded-left"
      value={value}
      onChange={onChange}
      id={text}
      placeholder={text}
      required
    />
  )
}

export default DateInput
