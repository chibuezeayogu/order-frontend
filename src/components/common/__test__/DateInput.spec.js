import React from 'react'
import { shallow } from 'enzyme'
import DateInput from '../DateInput'

const setUp = () => {
  const props = {
    value: '',
    onChange: jest.fn(),
    text: 'Booking Date'
  }

  return shallow(<DateInput {...props} />)
}

describe('DateInput Component', () => {
  it('should render without crashing', () => {
    const wrapper = setUp()
    const input = wrapper.find('input')

    expect(wrapper).toMatchSnapshot()
    expect(input).toHaveLength(1)
    expect(input.get(0).props.type).toBe('date')
    expect(input.get(0).props.required).toBeTruthy()
    expect(input.get(0).props.placeholder).toBe('Booking Date')
  })
})
