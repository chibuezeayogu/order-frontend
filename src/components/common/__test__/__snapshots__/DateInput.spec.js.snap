// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput Component should render without crashing 1`] = `
<input
  className="form-control rounded-left"
  id="Booking Date"
  onChange={[MockFunction]}
  placeholder="Booking Date"
  required={true}
  type="date"
  value=""
/>
`;
